syntax = "proto3";

import "DTO.proto";

package SSDTO;

message FriendInfo{
    bytes ssid = 1;
    bool is_group = 2;
    bytes nickname = 3;
    bytes remark = 4;
}

message GetFriendList_DTO{
    bytes request_ssid = 1;
    repeated FriendInfo friend_infos = 2;
    bytes ip = 3;
    SSDTO.Business_Type type = 4;
}

message AddFriend_DTO{
    FriendInfo sender = 1;
    FriendInfo receiver = 2;
    bytes is_accept = 3;
    bytes msg = 4;
    bytes time = 5;
    bytes ip = 6;
    SSDTO.Business_Type type = 7;
}

message Search_DTO{
    bytes request_ssid = 1;
    bytes key_word = 2;
    repeated FriendInfo search_infos = 3;
    int32 page_size = 4;
    int32 res_size = 5;
    bytes ip = 6;
    SSDTO.Business_Type type = 7;
}